xbmc/xbmc.cpp

```diff
@@ -53,7 +53,7 @@ extern "C" int XBMC_Run(bool renderGUI)
   g_RBP.LogFirmwareVerison();
 #endif
 
-  if (renderGUI && !g_application.CreateGUI())
+  if (renderGUI && !g_application.IsHeadless() && !g_application.CreateGUI())
```

Comment:
Now I found out why I had trouble with this one. Check main.cpp, renderGUI = true and appparam parses it and sets it elsewhere.

So basically it feels like there is a bit of a clash in the design here, you can either set it through param _or_ via renderGUI = false (which is used in kodi shared library for those that wants it)

However, we should probably do a g_application.SetHeadless(!renderGUI); in xbmc.cpp?
Not sure how we could refactor it otherwise to make it nice, would love suggestions though